dev/NavigationView/NavigationView.cpp

```diff
@@ -1779,10 +1779,90 @@ void NavigationView::OnNavigationViewItemKeyDown(const winrt::IInspectable& send
     }
 }
 
+void NavigationView::ArrowKeyNavigationPolyfill(const winrt::NavigationViewItem& nvi, const winrt::KeyRoutedEventArgs& args)
+{
+    switch (args.Key())
+    {
+    case winrt::VirtualKey::Down:
+        if (auto const ir = GetParentItemsRepeaterForContainer(nvi))
+        {
+            if (ir != m_topNavRepeater.get())
+            {
+                auto const indexOfCurrentElement = GetIndexFromItem(ir, nvi);
+                if (indexOfCurrentElement >= 0)
+                {
+                    if (FocusNextFocusableElement(ir, indexOfCurrentElement))
+                    {
+                        args.Handled(true);
+                    }
+                }
+            }
+        }
```

Comment:
lots of duplication between the Up, down and the next/previous blocks. I'm curious if you can reduce that by refactoring it like so

case down:
             if(auto next =  FindNextFocusibleElement(nvi, direction::next))
              {
                   SetFocus(next);
                   handled..
              }
...

Reply:
agreed, looks refactorable.

---
In reply to: [367157704](https://github.com/microsoft/microsoft-ui-xaml/pull/1850#discussion_r367157704) [](ancestors = 367157704)